onnx/defs/math/defs.cc

```diff
@@ -3660,4 +3660,534 @@ ONNX_OPERATOR_SET_SCHEMA(
+                const auto signal_ndim = 1;
+
+                // Add 1 dimensions to the rank for the batch size dimension
+                auto real_inputs_ndim = signal_ndim + batch_ndim;
+                // Add 2 dimensions to the rank for the batch size dimension AND component dimension
+                auto complex_inputs_ndim = signal_ndim + batch_ndim + component_ndim;
+
+                auto& input_shape = getInputShape(ctx, 0);
+                auto dim_size = static_cast<int64_t>(input_shape.dim_size());
+                auto is_real = dim_size == real_inputs_ndim;
+                auto is_complex = dim_size == complex_inputs_ndim;
+
+                ONNX_NAMESPACE::TensorShapeProto result_shape_proto = input_shape;
+                
+                bool is_onesided = static_cast<bool>(getAttribute(ctx, "onesided", 0));
+                if (is_onesided) {
+                    // Since signal_ndim = 1, and multidimensional DFT is not supported,
+                    // only the single signal dim (1) needs to be updated
+                    auto n_fft = input_shape.dim(1).dim_value();
+                    result_shape_proto.mutable_dim(1)->set_dim_value((n_fft >> 1) + 1);
+                }
+        
+                if (dim_size == real_inputs_ndim) {
+                    // For real input: the output is same shape, but with extra dim to represent complex output (real/imaginary)
+                    result_shape_proto.add_dim()->set_dim_value(2);  
+                } else if (dim_size == complex_inputs_ndim) {
+                    // For complex input: do nothing
+                } else {
+                    fail_shape_inference(
+                        "The input_shape must [batch_size][n_fft] for real values or [batch_size][n_fft][2] for complex values.")
+                }
+
+                updateOutputShape(ctx, 0, result_shape_proto);
+            }));
+
+static const char* IDFT_ver13_doc =
+    R"DOC(Computes the inverse discrete Fourier transform of input.)DOC";
+
+    ONNX_OPERATOR_SET_SCHEMA(IDFT,
+        13,
+        OpSchema()
+            .SetDoc(IDFT_ver13_doc)
+            .Input(
+                0,
+                "input",
+                "Input tensor representing a real or complex valued signal. "
+                "For real input, the following shape is expected: [batch_size][n_fft]. "
+                "For complex input, the following shape is expected: [batch_size][n_fft][2], where " 
+                "[batch_size][n_fft][0] represents the real component and [batch_size][n_fft][1] represents the imaginary component of the signal.",
+                "T",
+                OpSchema::Single,
+                true,
+                1,
+                OpSchema::NonDifferentiable)
+            .Output(
+                0,
+                "output",
+                "The inverse 1-dimensional discrete Fourier transform of the input. "
+                "The output tensor has the same shape and type as the input.",
```

Comment:
This does not seem consistent with the shape-inference function below, which may add an extra dimension for complex output.